contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,202 @@
+
+> npm install -g @angular/cli
+
+Running the above command installs the Angular CLI and gives you more control over your project. It also enforces "best practices" in terms of code quality.
+With Angular CLI developers can generate Angular files, execute applications and even run end to end tests.
+
+[source](https://www.amadousall.com/why-you-should-use-angular-cli/)
+
+### What does an Angular app look like?
+
+Let’s dive into some code, starting with a little hello world example. All Angular apps start in HTML.
+
+```
+<!doctype html>
+<html>
+  <head>
+    <!-- The various &lt;script&gt; tags needed to load Angular -->
+    <!-- For a full list, see https://angular.io/docs/ts/latest/quickstart.html -->
+  </head>
+  <body>
+    <my-app></my-app>
+  </body>
+</html>
+```
+
+The one unique thing in the above example is the `<my-app>` element, as that’s not an element you’d regularly use in a web-based application. Angular’s goal has always been to extend the vocabulary of HTML, and it does so by allowing you do define your own tags.
+These custom tags are known as components.
+
+Let's now look at some TypeScript code that defines the `<my-app>` component.
+
+```
+import { Component } from "@angular/core";
+
+@Component({
+  selector: "my-app",
+  template: `
+    <h1>Hello World</h1>
+  `
+})
+export class AppComponent {}
+```
+
+In Angular we use the `@Component` tag, which is known as a decorator, to mark classes that should be considered elements that can be used in your HTML markup. You have the ability to pass that `@Component` properties to describe the element.
+
+* The `selector` property determines that tag’s name when typed in HTML. The use of selector: "my-app" is how Angular knows what to do when it sees a `<my-app>` tag in HTML.
+
+* The `template` property controls what HTML gets rendered when this component is used.
+
+A visual to summarize the above :
+
+![Component](https://angular.io/generated/images/guide/architecture/component-databinding.png)
+
+### Why not to use Angular
+
+Like any other framework/library, Angular has its share of disadvantages.
+
+1. **Steep Learning curve** - Angular has many topics to learn, starting from basic ones such as directives, modules, decorators, components, services, dependency injection, pipes, templates and most importantly TypeScript. 
+The sheer number of new concepts is confusing to newcomers.
+
+2. **Opiniated Framework** - Angular tells developers how to structure their application. While libraries like React allow engineers the freedom to choose any third party library to integrate into their application.
+
+3. **Performance** - In the past, Dynamic applications written in Angular did not always perform that well even though improvements have been made recently. Complex SPAs could be laggy and inconvenient to use due to their size. 
```

Comment:
> In the past, Dynamic applications written in Angular

How long ago was that? During AngularJS? Angular2? Need to be more specific here.